ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/CodeNodeDataRequest.java

```diff
@@ -46,7 +46,9 @@ protected void doPersist(final Updater updater) {
 
   @Override
   public Optional<Bytes> getExistingData(final WorldStateStorage worldStateStorage) {
-    return worldStateStorage.getCode(getHash(), accountHash.orElse(Hash.EMPTY));
+    return worldStateStorage
+        .getCode(getHash(), accountHash.orElse(Hash.EMPTY))
+        .filter(bytes -> Hash.hash(bytes).compareTo(getHash()) == 0);
```

Comment:
if I understand well, we can store empty codes, and if yes why?

Reply:
no we cannot the Hash empty is for the accountHash not the code. 🤔 And it's not possible in this part of the code to have an empty accountHash. the best will be to do .orElseThrow instead of that. I will update the code